packages/data-stores/src/auth/controls.ts

```diff
@@ -0,0 +1,49 @@
+/**
+ * External dependencies
+ */
+import { createRegistryControl } from '@wordpress/data';
+import { stringify } from 'qs';
+
+/**
+ * Internal dependencies
+ */
+import { wpcomRequest, WpcomClientCredentials } from '../utils';
+import { FetchAuthOptionsAction, FetchWpLoginAction } from './actions';
+import { STORE_KEY } from './constants';
+
+export function createControls( clientCreds: WpcomClientCredentials ) {
+	return {
+		SELECT_USERNAME_OR_EMAIL: createRegistryControl( registry => () => {
+			return registry.select( STORE_KEY ).getUsernameOrEmail();
+		} ),
+		FETCH_AUTH_OPTIONS: async ( { usernameOrEmail }: FetchAuthOptionsAction ) => {
+			const escaped = encodeURIComponent( usernameOrEmail );
+
+			return await wpcomRequest( {
+				path: `/users/${ escaped }/auth-options`,
+				apiVersion: '1.1',
+			} );
+		},
+		FETCH_WP_LOGIN: async ( { action, params }: FetchWpLoginAction ) => {
+			const response = await fetch(
+				// TODO Wrap this in `localizeUrl` from lib/i18n-utils
+				'https://wordpress.com/wp-login.php?action=' + encodeURIComponent( action ),
+				{
+					method: 'POST',
+					credentials: 'include',
+					headers: {
+						Accept: 'application/json',
+						'Content-Type': 'application/x-www-form-urlencoded',
+					},
+					body: stringify( {
+						remember_me: true,
+						...clientCreds,
+						...params,
+					} ),
+				}
+			);
+
+			return await response.json();
+		},
+	};
```

Comment:
A shared, plain `fetch` control that sets typical defaults like you've done here would also be nice 🙂 Most of our usage of `apiFetch` should either be wpcom-proxy-request or a plain fetch, I believe continued usage of the `apiFetch` singleton will only become more complicated and I'd prefer to migrate away from it.